src/libsync/discovery.cpp

```diff
@@ -539,18 +539,18 @@ void ProcessDirectoryJob::processFileAnalyzeRemoteInfo(
             } else {
                 item->_instruction = CSYNC_INSTRUCTION_SYNC;
             }
-        } else if (dbEntry._modtime <= 0 && serverEntry.modtime > 0) {
+        } else if ((dbEntry._modtime <= 0 && serverEntry.modtime > 0) || dbEntry._modtime != serverEntry.modtime) {
             item->_direction = SyncFileItem::Down;
             item->_modtime = serverEntry.modtime;
             item->_size = sizeOnServer;
-            if (serverEntry.isDirectory) {
-                ENFORCE(dbEntry.isDirectory());
-                item->_instruction = CSYNC_INSTRUCTION_UPDATE_METADATA;
-            } else if (!localEntry.isValid() && _queryLocal != ParentNotChanged) {
-                // Deleted locally, changed on server
-                item->_instruction = CSYNC_INSTRUCTION_NEW;
-            } else {
-                item->_instruction = CSYNC_INSTRUCTION_SYNC;
+            item->_instruction = CSYNC_INSTRUCTION_UPDATE_METADATA;
+            if (dbEntry._modtime != serverEntry.modtime && dbEntry._etag == serverEntry.etag) {
+                qCWarning(lcDisco()) << "server modification time has been modified without changing etag: force full discovery";
+                _forceRemoteDiscovery = true;
+            }
+            if (dbEntry._modtime != serverEntry.modtime && dbEntry._etag == serverEntry.etag) {
```

Comment:
@mgallien Why do you need to repeat the same logic here?

Reply:
I will check

Reply:
thanks that part got duplicated during rebase
thanks for catching it